besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -1545,7 +1559,14 @@ private PrivacyParameters privacyParameters() {
       }
     }
 
-    return privacyParametersBuilder.build();
+    final PrivacyParameters privacyParameters = privacyParametersBuilder.build();
+
+    if (isPrivacyEnabled) {
+      preSynchronizationTaskRunner.addTask(
```

Comment:
Because part of the task is to evaluate the different triggering conditions. And one of the conditions is: you have a private db that needs migration but you don't have the migratePrivateDatabase flag enabled.